flash/image/classification/data.py

```diff
@@ -207,6 +471,101 @@ def from_data_frame(
         transform_kwargs: Optional[Dict] = None,
         **data_module_kwargs: Any,
     ) -> "ImageClassificationData":
+        """Load the :class:`~flash.image.classification.data.ImageClassificationData` from pandas DataFrame objects
+        containing image files and their corresponding targets.
+
+        Input images will be extracted from the ``input_field`` in the DataFrame.
+        The supported file extensions are: ``.jpg``, ``.jpeg``, ``.png``, ``.ppm``, ``.bmp``, ``.pgm``, ``.tif``,
+        ``.tiff``, ``.webp``, and ``.npy``.
+        The targets will be extracted from the ``target_fields`` in the DataFrame and can be in any of our
+        :ref:`supported classification target formats <formatting_classification_targets>`.
+        To learn how to customize the transforms applied for each stage, read our
+        :ref:`customizing transforms guide <customizing_transforms>`.
+
+        Args:
+            input_field: The field in the DataFrames containing the image files.
+            target_fields: The field or list of fields in the DataFrames containing the targets.
+            train_data_frame: The pandas DataFrame to use when training.
+            train_images_root: The root directory containing train images.
+            train_resolver: Optionally provide a function which converts an entry from the ``input_field`` into an image
+                file path.
+            val_data_frame: The pandas DataFrame to use when validating.
+            val_images_root: The root directory containing validation images.
+            val_resolver: Optionally provide a function which converts an entry from the ``input_field`` into an image
+                file path.
+            test_data_frame: The pandas DataFrame to use when testing.
+            test_images_root: The root directory containing test images.
+            test_resolver: Optionally provide a function which converts an entry from the ``input_field`` into an image
+                file path.
+            predict_data_frame: The pandas DataFrame to use when predicting.
+            predict_images_root: The root directory containing predict images.
+            predict_resolver: Optionally provide a function which converts an entry from the ``input_field`` into an
+                image file path.
+            train_transform: The :class:`~flash.core.data.io.input_transform.InputTransform` type to use when training.
+            val_transform: The :class:`~flash.core.data.io.input_transform.InputTransform` type to use when validating.
+            test_transform: The :class:`~flash.core.data.io.input_transform.InputTransform` type to use when testing.
+            predict_transform: The :class:`~flash.core.data.io.input_transform.InputTransform` type to use when
+              predicting.
+            input_cls: The :class:`~flash.core.data.io.input.Input` type to use for loading the data.
```

Comment:
without click cross-link, no idea what it means

Reply:
Yeah, would be good to add a page on how to use the `input_cls` argument for customizing data loading. Will do in a follow-up :smiley: